axonserver/src/main/java/io/axoniq/axonserver/message/query/QueryDispatcher.java

```diff
@@ -189,7 +190,7 @@ public void query(SerializedQuery serializedQuery, Authentication principal,
             if (handlers.isEmpty()) {
                 interceptedCallback.accept(QueryResponse.newBuilder()
                                                         .setErrorCode(ErrorCode.NO_HANDLER_FOR_QUERY.getCode())
-                                                        .setMessageIdentifier(query.getMessageIdentifier())
```

Comment:
The messageIdentifier was indeed wrongly set to the request identifier. However, each message should also have a `MessageIdentifier`. It can be set to `UUID.randomUUID().toString()`

Reply:
Sounds good. I'll reimplement it.